.travis.yml

```diff
@@ -1,9 +1,20 @@
 language: go
 go:
-  - "1.11"
+- '1.11'
 os:
-  - linux
-  - osx
-  - windows
-script:
-  - go test -v ./...
\ No newline at end of file
+- linux
+- osx
+- windows
+script: go test -v ./...
+
+deploy:
+  script: make release
+  provider: releases
+  api_key:
+    secure: sofGKsRoyvTdPeSAn1rNcsIfU9+35RJLus3CtiRUiC7W0JBk5gb40Ij3blHxkD2n4qXDXaYelvEG7RvaRmI/Ow/Zrun3/bJzdSnAhnxM81TZ8xLOJQquKz+esgkiJ2rLBYO451+uIvRwmdH50945pkraO3qsASAvBOTtoA+20QwSDMEvT1Q03Z7wj+dbvljnfXrf2C1vXPczZnZlI9vts85TVCzsn2K/kBpxLgneGX8EFc+qTWqUchd7wPi1VvPOwkr8GbCz8hUbcXDmR44e4m7EsSNNoiCeKybqtVtXlNzsFGbXhP9+dCQj/oLw4yAYeSpVnHWSxaQNipgLfnw05yqnQW/dBtRnDshwcyx3u7u0C/owFhS5/LWOC0VUF0Doyq89XtJfDsMO1jmahs6k/iZW/i3ke/GrxK5pZ7H2kbtFmB58kS05XeMdffXyfak4bPyaYJwZM5gkoQhhB6WJ0oUAvfWKMC6a2ScLXtF6qgLmKERah8nkbIkT+uD/+pVmajCUdhDVcQZf1H2sBOFo5RrX2kohuGMoHb6twIT87CBuqox7J1oxJNsZAo1Hg8Vp1iscxNfUo5AJn/JoNAury5jDsCvZiT8bpntZZ8QaPwmf1kgUpQrt4X6lXFelv0bhap6KfNHsg5ea5Ts+TB30WBQ0l+42EQLm+X5WZV29Zpw=
+  file_glob: true
+  file: skuid_*
+  skip_cleanup: true
+  on:
```

Comment:
What triggers this deploy script to be run? On a tag being pushed? I think one extension to this that we need to tweak is to do tag-based versioning. Right now there are like 3 places in `skuid-cli` code where the version (e.g. 0.3.11) is hard-coded. I was hoping that we could be able to have this grabbed from a tag and used to basically build-in the version number into the binary distributable.

Potentially that's an extension to this story here, but I am just checking to see what would actually initiate the deploy script --- seems relevant to discuss how that ties into specifying the version number. want to make this as seamless as possible.

Reply:
My vision is to work toward 100% tagged SemVers driving the whole thing. For the moment I need to test this integration from a tag. I'll be submitting a followup shortly to integrate semver tags into the actual codebase.

Reply:
So, the `make release` script does a `git tag` operation --- won't this always fail, currently, because there's preexisting tags?

Reply:
It will indeed. That's a miss on my part - removed that line in a followup commit. Test plan for following up on this is to submit a tag to the repo both through a git interface and the release interface to observe the TravisCI behavior - the docs explain sort of how it works, but until I get to watch it in action I am speculating to some degree.